version:  '3'
services:
  datascience-notebook:
      image: jupyter/datascience-notebook
      volumes:
        - ${LOCAL_WORKING_DIR}:/home/jovyan/work
        - ${LOCAL_DATASETS}:/home/jovyan/work/datasets
        - ${LOCAL_MODULES}:/home/jovyan/work/modules
      container_name: jupyter_notebook
      networks:
        - traefik
      labels:
        - "traefik.enable=true"
        - "traefik.http.services.jupyter.loadbalancer.server.port=8888"
        - "traefik.http.routers.jupyter.rule=Host(`jupyter.snipermonkey.de`)"
        - "traefik.http.routers.jupyter.entrypoints=web"
        - "traefik.http.routers.jupyter.tls=true"
        - "traefik.http.routers.jupyter.middlewares=restricted-access@file"
      restart: unless-stopped
networks:
  traefik:
    external: true